lib/models/host.rb

```diff
@@ -119,7 +119,8 @@ def api_format
   end
 
   def define_cpus
-    cpus.map { |e| e[:cores] = cpu_cores }
+    total_cores = (cpu_cores.present? && cpus.count > 0) ? (cpu_cores/cpus.count) : 0
```

Comment:
What do you think of using:
total_cores = (cpus && cpus.count > 0) ? (cpu_cores.to_i/cpus.count.to_i) : 0